.github/workflows/php.yml

```diff
@@ -0,0 +1,33 @@
+name: PHP Composer
+
+on:
+  push:
+    branches: [ "**" ]
+  pull_request:
+    branches: [ "main" ]
+
+permissions:
+  contents: read
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Cache Composer packages
+        id: composer-cache
+        uses: actions/cache@v3
+        with:
+          path: vendor
+          key: ${{ runner.os }}-php-${{ hashFiles('**/composer.lock') }}
```

Comment:
If there's not going to be a `composer.lock` file, not sure if that will work to retrieve cache elements.

we used the `composer.json` for the purpose and the php version we built dependencies & tested on, example -> https://github.com/upmind-automation/provision-provider-domain-names/blob/main/.github/workflows/static-analysis.yml